app/plugins/Wallet/js/lifecycle.js

```diff
@@ -46,8 +48,20 @@ function updateAddrTxn(addr) {
 	}, 'update-history');
 }
 
-// Update wallet summary in header
+// Update transaction history and addresses
 addResultListener('update-status', function(wallet) {
+	refreshRate = finalRefreshRate; // slow down after first successful call
+
+	// Show correct lock status.
+	// TODO: If the wallet is encrypted, prompt with a pw.
```

Comment:
password prompt should not appear until the user hits the 'encrypt' button. Standard instinct when dealing with app popups is to TL;DR and close them, then people don't know their password.